### PR TITLE
Docker build env vars

### PR DESCRIPTION
**Pull request by npanpaliya**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/master/CONTRIBUTING.md)?
- [x] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/master/doc/)?
- [x] Did you write any [tests](https://github.com/open-ce/open-ce/blob/master/tests/) to validate this change?  

## Description

Fixes https://github.com/open-ce/open-ce/issues/127.

I'll update the documentation while this is being reviewed.

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/master/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

**Comment by npanpaliya** [comment]
I've updated the documentation as reported in https://github.com/open-ce/open-ce/issues/291.

File: open-ce/docker_build.py [hunk 1]
```diff
@@ -140,8 +154,8 @@ def build_in_container(image_name, output_folder, arg_strings):
     if os.path.isdir(LOCAL_FILES_PATH):
         _copy_to_container(LOCAL_FILES_PATH, HOME_PATH, container_name)
 
-    _start_container(container_name)
```
Comment (jayfurmanek) [inline open-ce/docker_build.py:157]: looks like an extra line was added here that's not needed.

**Comment by jayfurmanek** [comment]
This looks good! It will need to account for the new conflicts.
A good way to test this function would be trying to pass in CUDA_VISIBLE_DEVICES.

**Comment by bethune-bryant** [comment]
Are there any other docker options that people may want to use?

Should we generalize this to just be `--docker_build_args`, and then the user could do:
```
./open-ce/open-ce build env --docker_build --docker_build_args="--build-arg ENV1=test1 --build-arg ENV2=test2" envs/pytorch-env.yaml
```
And they could also do more generic things like:
```
./open-ce/open-ce build env --docker_build --docker_build_args="--build-arg ENV1=test1 --cpuset-cpus 0,1" envs/pytorch-env.yaml
```

@bnemanich @jayfurmanek

**Comment by bnemanich** [comment]
Yes, I think what @bethune-bryant mentioned makes sense.

**Comment by npanpaliya** [comment]
@bethune-bryant - This makes sense to me too. I'll incorporate this.

**Comment by npanpaliya** [comment]
I'll update the documentation once the implementation is approved.

**Comment by bethune-bryant** [comment]
This looks good to me now!

File: open-ce/docker_build.py (hunk 1 above)
Comment (npanpaliya) [reply open-ce/docker_build.py:175]: Corrected.

**Review by bnemanich — APPROVED** [review]

**Review by cdeepali — APPROVED** [review]

**Pull request merged**